src/WebJobs.Extensions.ApiHub/Listener/ApiHubListener.cs

```diff
@@ -86,5 +133,83 @@ private Task OnFileWatcher(IFileItem file, object obj)
                 _poll = null;
             }
         }
+
+        private async Task<string> GetLastPollStatusAsync()
+        {
+            try
+            {
+                var apiHubBlob = GetBlobReference();
+
+                var statusLine = await apiHubBlob.DownloadTextAsync();
+                ApiHubStatus status;
+                using (StringReader stringReader = new StringReader(statusLine))
+                {
+                    status = (ApiHubStatus)_serializer.Deserialize(stringReader, typeof(ApiHubStatus));
+                }
+                return status.PollUrl;
+            }
+            catch (StorageException exception)
+            {
+                if (exception.RequestInformation == null ||
+                    exception.RequestInformation.HttpStatusCode != 404)
+                {
+                    var errorText = string.Format("Encountered a storage exception while getting the next poll status from the blob storage. Path: {0}, Message: {1} ", this._folderSource.Path, exception.Message);
+                    _trace.Error(errorText);
+                }
+                return null;
+            }
+            catch (Exception e)
+            {
+                var errorText = string.Format("Encountered an exception while getting the next poll status from the blob storage. Path: {0}, Message: {1} ", this._folderSource.Path, e.Message);
+                _trace.Error(errorText);
+                return null;
+            }
+        }
+
+        private async Task SetNextPollStatusAsync(string nextPoll)
+        {
+            try
+            {
+                var apiHubBlob = GetBlobReference();
+
+                ApiHubStatus status = new ApiHubStatus
+                {
+                    PollUrl = nextPoll
+                };
+
+                string statusLine;
+                using (StringWriter stringWriter = new StringWriter())
+                {
+                    _serializer.Serialize(stringWriter, status);
+                    statusLine = stringWriter.ToString();
+                }
+
+                await apiHubBlob.UploadTextAsync(statusLine);
+            }
+            catch (Exception e)
+            {
+                var errorText = string.Format("Encountered an exception while setting the next poll status to the blob storage. Path: {0}, Message: {1} ", this._folderSource.Path, e.Message);
```

Comment:
would be good not to repeat this same literal 3+ times. recommend moving to resource file